Library/Homebrew/version.rb

```diff
@@ -379,3 +392,29 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :commit
+
+  def initialize(val)
+    unless val.respond_to?(:to_str)
```

Comment:
I'm not sure. It seems that it's not needed, but it ensures that if someone calls `HeadVersion.new`, it'll fail if `val` doesn't respond to `to_str`.

How about this?
```ruby
def initialize(val)
  super
  @commit = @version[/^HEAD-(.+)$/, 1]
end
```

Also we can add  `private_class_method :new` to `HeadVersion` and don't do the `to_str` check, but again, I'm not sure this is a good idea.